core/arch/arm/encode.c

```diff
@@ -2766,7 +2766,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     decode_info_init_for_instr(&di, instr);
-    di.opcode = instr_get_opcode(instr);
+    di.opcode = instr->opcode;
```

Comment:
Ah, the original problem... I think it's that `INSTR_RAW_BITS_VALID` gets cleared by the use of `instr_get_opcode`, perhaps by a call to `instr_reuse`.